### PR TITLE
LLM: Add baichuan2 gpu spec example

### PR DESCRIPTION
**Pull request by cyita**
## Description

Add baichuan2 gpu speculative example

### 4. How to test?
- [x] local test baichuan 7b/13b with transformers 4.31.0
- [x] local test baichuan 7b/13b with transformers 4.36.0

### CODE REVIEW

File: python/llm/example/GPU/Speculative-Decoding/baichuan2/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+### 2. Configures OneAPI environment variables
+```bash
+source /opt/intel/oneapi/setvars.sh
+```
+
+### 3. Run
+
+For optimal performance on Intel Arc™ A-Series Graphics and Intel Data Center GPU Flex Series, it is recommended to set several environment variables.
+```bash
+export USE_XETLA=OFF
```
Comment (rnwang04) [inline python/llm/example/GPU/Speculative-Decoding/baichuan2/README.md:27]: we need to remove these.

File: python/llm/example/GPU/Speculative-Decoding/baichuan2/speculative.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import numpy as np
+
+
+torch.nn.Linear.reset_parameters = lambda x: None
+seed=42
+torch.manual_seed(seed)
+np.random.seed(seed)
+
+# you could tune the prompt based on your own model,
+BAICHUAN2_PROMPT_FORMAT = "<human>{prompt} <bot>"
```
Comment (rnwang04) [inline python/llm/example/GPU/Speculative-Decoding/baichuan2/speculative.py:31]: remove this?

**Review by rnwang04 — APPROVED** [review]
others LGTM
